modules/custom/lua/helm_claim.lua

```diff
@@ -0,0 +1,60 @@
+-----------------------------------
+-- HELM Claim
+-----------------------------------
+-- Gathering Points are "claimed" for x seconds upon use
+-- Prevents players from "stealing" a point currently in use
+-----------------------------------
+require("modules/module_utils")
+-----------------------------------
+local m = Module:new("helm_claim")
+
+local claimSettings = {
+    MESSAGE = "Another player is currently using this.",
+    CLAIM_DURATION = 10, -- Seconds
+    POINT_AVAILABLE = "POINT_AVAILABLE",
+    POINT_CLAIMED = "POINT_CLAIMED"
+}
+
+local helmTools = {
+    [xi.helm.type.HARVESTING] = 1020,
+    [xi.helm.type.EXCAVATION] = 605,
+    [xi.helm.type.LOGGING] = 1021,
+    [xi.helm.type.MINING] = 605,
+}
+
+if xi.settings[m.name] then
+    if xi.settings[m.name].MESSAGE then
+        claimSettings.MESSAGE = xi.settings[m.name].MESSAGE
+    end
+
+    if xi.settings[m.name].CLAIM_DURATION then
+        claimSettings.CLAIM_DURATION = xi.settings[m.name].CLAIM_DURATION
+    end
+end
+
+m:addOverride("xi.helm.onTrade", function(player, npc, trade, helmType, csid, func)
+    if os.time() < npc:getLocalVar(claimSettings.POINT_AVAILABLE) and player:getID() ~= npc:getLocalVar(claimSettings.POINT_CLAIMED) then
```

Comment:
potential issue: Could block players from completing quests that involve trading to mining points (I think drg unlock one?) because of players mining.
suggestion: Check this after checking `trade:hasItemQty(helmTools[helmType], 1)`

Reply:
For the case mentioned, there's a [quest check](https://github.com/AirSkyBoat/AirSkyBoat/blob/staging/scripts/zones/Maze_of_Shakhrami/npcs/Excavation_Point.lua#L13) before `xi.helm.onTrade` is called. I guess this will be moved to quest file after rerwrite but should still be fine. So that shouldn't be affected.

For the others, they currently reside in the HELM global, which kind of messy, and they should probably be moved. Will have to be another PR. Either way, I don't think it's a *huge* issue as it's only a tiny inconvenience in some rare edge cases. Progress won't be prevented, you just have to find one not being used, or ask nicely. :)